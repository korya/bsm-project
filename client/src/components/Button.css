/* Probably could use a standard bootstrap or material-design button.
 * But why not to use a non-standard one?
 * Based on https://codepen.io/andybelldesign/pen/Vxpjvo
 */

button.button {
  display: inline-block;
  border: none;
  padding: 1rem 2rem;
  margin: 0;
  text-decoration: none;
  background: #0069ed;
  color: #ffffff;
  font-family: sans-serif;
  font-size: 1rem;
  line-height: 1;
  cursor: pointer;
  text-align: center;
  transition: background 250ms ease-in-out, transform 150ms ease;
  -webkit-appearance: none;
  -moz-appearance: none;
}

button.button:hover {
  background: #0053ba;
}

button.button:focus {
  outline: 0;
}

button.button:active {
  transform: scale(0.95);
}
