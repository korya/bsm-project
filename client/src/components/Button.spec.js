import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './Button';

test('renders correctly', () => {
  expect(
    renderer.create(<Button />).toJSON(),
  ).toMatchSnapshot();
  expect(
    renderer.create(<Button onClick={() => {}}>Label</Button>).toJSON(),
  ).toMatchSnapshot();
});

test('renders children correctly', () => {
  const wrapper = shallow(<Button />);
  expect(wrapper.html()).toMatchSnapshot();
});

test('calls onClick prop handler', () => {
  const clickHandler = jest.fn();
  const wrapper = shallow(<Button onClick={clickHandler} />);
  wrapper.find('button').simulate('click');
  expect(clickHandler).toHaveBeenCalledTimes(1);
});
