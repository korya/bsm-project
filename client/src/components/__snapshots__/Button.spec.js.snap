// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders children correctly 1`] = `"<button class=\\"button\\"></button>"`;

exports[`renders correctly 1`] = `
<button
  className="button"
/>
`;

exports[`renders correctly 2`] = `
<button
  className="button"
  onClick={[Function]}
>
  Label
</button>
`;
